monitor/src/config.rs

```diff
@@ -20,11 +25,17 @@ impl Config {
 
         let config: Config = serde_yaml::from_str(&mut contents).expect("Failed to parse YAML");
 
-        // Verify that the actions file exists
-        if !Path::new(&config.actions_path).exists() {
-            panic!("The actions file '{}' does not exist.", config.actions_path);
+        // Verify that the actions files exist for each configuration
+        for single_config in &config.configurations {
+            Config::exists_actions_file(single_config);
```

Comment:
```suggestion
            if !Path::new(&config.actions_path).exists() {
            return Err( !format("File {} not found" , config.actions_path) )
        }
```

Then you will catch the error upward in the code.